File: leet_code/linked_lists/remove_nth_node.js
const removeNthNode = (head, n) => {
  if (!head) return head;

  let length = 0;
  let cur = head;

  while (cur) {
    length += 1;
    cur = cur.next;
  }

  if (n === length) {
    head = head.next;
    return head;
  }

  let count = 1;
  cur = head;

  while (count < (length - n)) {
    cur = cur.next;
    count += 1;
  }
  cur.next = cur.next.next;
  return head;
};

File: leet_code/linked_lists/remove_nth_node.rb
def remove_nth_node(head, n)
  return head if !head

  length, cur = 0, head

  while cur
    length += 1
    cur = cur.next
  end

  return head = head.next if n == length

  count, cur = 1, head
  
  while count < (length - n)
    cur = cur.next
    count += 1
  end
  cur.next = cur.next.next
  head
end
